src/features/auth/components/Register/RegisterForm.tsx

```diff
@@ -22,7 +22,10 @@ const fields = [
 
 export const RegisterForm = () => {
   const { t } = useTranslation();
-  const { onRegister } = useSession();
+  const { register } = useSession();
+  const toast = useToast();
+
+  const [isSigningUp, setIsSigningUp] = useState(false);
```

Comment:
You could use redux/redux toolkit and use async thunk which would help you handle this

Reply:
I prefer to avoid using third-party libraries to handle state unless it's necessary. React's built-in solutions are usually just as good.